src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -61,36 +61,54 @@ public void Initialize(ExtensionConfigContext context)
 
         public void ValidateBinding(RabbitMQAttribute attribute, Type type)
         {
+            string connectionString = Utility.FirstOrDefault(attribute.ConnectionStringSetting, _options.Value.ConnectionString);
             string hostName = Utility.FirstOrDefault(attribute.HostName, _options.Value.HostName);
-            string queueName = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName);
-
-            if (string.IsNullOrEmpty(hostName))
-            {
-                throw new InvalidOperationException("RabbitMQ host name is missing");
-            }
+            string queueName = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName) ?? throw new InvalidOperationException("RabbitMQ queue name is missing");
 
-            if (string.IsNullOrEmpty(queueName))
+            if (!string.IsNullOrEmpty(hostName) && string.IsNullOrEmpty(connectionString) && !hostName.Equals("localhost", StringComparison.InvariantCultureIgnoreCase))

```

Comment:
Same here: Move the condition to check for localhost to utility and you can reuse that function in TriggerBindingProvider as well